pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/HttpURIDereferencerImpl.scala

```diff
@@ -12,13 +11,19 @@ class HttpURIDereferencerImpl(client: Client) extends URIDereferencer {
   override def dereference(uri: URI): IO[URIDereferencerError, String] = {
     val result: ZIO[Client, URIDereferencerError, String] = for {
       response <- Client.request(uri.toString).mapError(t => ConnectionError(t.getMessage))
-      body <- response match
-        case Response(Status.Ok, _, body, _, None) =>
-          body.asString.mapError(t => UnexpectedError(t.getMessage))
-        case Response(Status.NotFound, _, _, _, None) =>
-          ZIO.fail(ResourceNotFound(uri))
-        case Response(_, _, _, _, httpError) =>
-          ZIO.fail(UnexpectedError(s"HTTP response error: $httpError"))
+      body <- response.status match {
+        case Status.Ok =>
+          response.body.asString.mapError(t => UnexpectedError(t.getMessage))
+        case Status.NotFound if !response.status.isError => ZIO.fail(ResourceNotFound(uri))
+        case _ if response.status.isError =>
+          val err = response match {
+            case Response.GetError(error) => Some(error)
+            case _                        => None
+          }
+          ZIO.fail(UnexpectedError(s"HTTP response error: $err"))
+        case _ =>
+          ZIO.fail(UnexpectedError("Unknown error"))
```

Comment:
A little confused by these 2 branches. When would the `response.status` not be error and we need return unexpected error.